src/workerd/util/sqlite.c++

```diff
@@ -1177,6 +1188,9 @@ sqlite3_stmt* SqliteDatabase::Query::getStatement() {
 }
 
 void SqliteDatabase::Query::beforeSqliteReset() {
+  // update the db stats before resetting the database
+  db.sqliteObserver.updateQueryStats(getRowsRead(), getRowsWritten());
```

Comment:
The asan error is happening because the statement could be dangling by the time we reach here. I am trying to see how I can fix this, but so far it does not look like there is a good way for us to figure this out.
We could store rowsRead and rowsWritten by the query and update it everytime we do any operation on the query and use it here, I am not 100% convinced I want to go that route yet.